### PR TITLE
Categorical color method and refactored tick formatting

### PR DESCRIPTION
**Pull request by ahwillia**

### CODE REVIEW

File: jetpack/chart/utils.py [hunk 1]
```diff
@@ -194,3 +249,40 @@ def setcolor(color='#444444', **kwargs):
     ax.set_ylabel(ax.get_ylabel(), color=color)
 
     return ax
+
+
+def categories_to_colors(data, color_cycle=None):
```
Comment (nirum) [inline jetpack/chart/utils.py]: 👍 super helpful function!!

File: jetpack/chart/utils.py [hunk 2]
```diff
@@ -116,6 +116,58 @@ def nospines(left=False, bottom=False, top=True, right=True, **kwargs):
 
     return ax
 
+@axwrapper
+def minimal_xticks(decimals=None, n_ticks=4, pad=True, **kwargs):
```
Comment (nirum) [inline jetpack/chart/utils.py:125]: This feels verbose... what about this kind of API?

```python
def min_xticklabels(bounds=None, n_ticks=4, **kwargs):
    if bounds is None:
        t0, t1 = ax.get_xlim()
    else:
        t0, t1 = bounds

    ticks = np.linspace(t0, t1, n_ticks)
    ax.set_xticks(ticks)
    ax.set_xticklabels([str(t0), *['' for _ in range(len(ticks)-2)], str(t1)])
    return ax
```

**Review by nirum — COMMENTED** [review]
I want to simplify your minlabels functions a bit

**Comment by nirum** [comment]
this PR has become too big :P

**Comment by ahwillia** [comment]
This PR has taken too long to review ;)

I'll close and reopen on a separate branch.

**Pull request closed without merging**